ArduCopter/control_safe_rtl.cpp

```diff
@@ -0,0 +1,136 @@
+#include "Copter.h"
+
+/*
+ * Init and run calls for Safe_RTL flight mode
+ *
+ * This code uses the SafeRTL path that is already in memory, and feeds it into WPNav, one point at a time.
+ * Once the copter is close to home, it will run a standard land controller.
+ */
+
+bool Copter::safe_rtl_init(bool ignore_checks)
+{
+    Vector3f current_pos;
+    if ((position_ok() || ignore_checks) && g2.safe_rtl.is_active() && ahrs.get_relative_position_NED_origin(current_pos)) {
+        // initialise waypoint and spline controller
+        wp_nav->wp_and_spline_init();
+
+        // if we're within 4m of the landing point, just skip right to pre-land-positioning
+        // otherwise, begin a normal safertl procedure from the start
+        if (HYPOT(current_pos, g2.safe_rtl.get_point(0)) <= 4.0f) {
+            current_pos[2] -= 2.0f; // go to the point two meters above the landing spot, then start landing.
```

Comment:
you're right.  I'm sure this is my mistake because I added the stopping-point stuff.  I think I'm just going to remove all this and let it do a regular SafeRTL even if it's within 4m of the target location.